include/pros/adi.hpp

```diff
@@ -32,12 +40,27 @@ class ADIPort {
 	 * reached:
 	 * ENXIO - The given value is not within the range of ADI Ports
 	 *
-	 * \param port
+	 * \param adi_port
 	 *        The ADI port number (from 1-8, 'a'-'h', 'A'-'H') to configure
 	 * \param type
 	 *        The configuration type for the port
 	 */
-	ADIPort(std::uint8_t port, adi_port_config_e_t type = E_ADI_TYPE_UNDEFINED);
+	ADIPort(std::uint8_t adi_port, adi_port_config_e_t type = E_ADI_TYPE_UNDEFINED);
+	
+	/**
+	 * Configures an ADI port on an adi expander to act as a given sensor type.
+	 *
+	 * This function uses the following values of errno when an error state is
+	 * reached:
+	 * ENXIO - The given value is not within the range of ADI Ports
+	 *
+	 * \param port
+	 *        The pair of the ADI port number (from 1-8, 'a'-'h', 'A'-'H') and the
+	 *  	  smart port number (from 1-22) to configure 
+	 * \param type
+	 *        The configuration type for the port
+	 */
+	ADIPort(ext_adi_port_pair_t port, adi_port_config_e_t type = E_ADI_TYPE_UNDEFINED);
```

Comment:
I'm not sure how much you want to favour correctness over readability, but constructors that can be called with a single argument should be marked as explicit. This prevents an integer from suddenly and mysteriously becoming an ADIPort with undefined type. 
Same goes for all the other C++ classes here.